### ISSUE TITLE
Unclear location of sequence descriptor strings

### ISSUE THREAD

**Issue by anadon** [dkopecek/libfasta#2]
Throughout the code, there is not a whole lot of documentation.  For FASTA_req_t, seq could have new lines or not from a reading of the documentation, but this is easy to find for certain.  What isn't is where the description strings are located.  The header sections appear to hold indexing information and not any form of description strings.  I can't be sure however, as the documentation is not complete enough.

**Comment by dkopecek** [comment]
I'll try to add documentation regarding this. I wasn't sure if anyone would be interested in this library at all, so I haven't invested much time in documenting everything. Anyway, I'll try to fix that in the coming days.

Any contributions are highly appreciated, thanks!

**Comment by anadon** [comment]
I'll be doing more tomorrow (after I can wake up at the crack of noon, that is).  I'm trying to roll out a new version of mummer, but the base code is garbage so I have to re-implement everything to not use 7k lines with macro dependencies and main() redefined.

**Comment by anadon** [comment]
I was looking through to see where potential slowdowns might be, and while I have some huge things on my end (n^2 stuff) I found that libfasta used *FILE stuff and not mmap.  From what I have read, using mmap has some advantages in memory usage (loading blocks needed) and tends to perform better.  Do you think making such a change would be relatively easy to libfasta?

**Comment by dkopecek** [comment]
Good idea. Don't hesitate to open a new ticket requesting this feature, so that it can be tracked.

**Comment by anadon** [comment]
Once I get my code adequately improved.  20 minutes and 8GB is too rich for my blood.

**Comment by anadon** [comment]
I don't understand you code well enough to make changes that don't break everything.  I need more documentation before I can help instead of recoding it all.
